tests/frontend/test_BoardGUI.py

```diff
@@ -0,0 +1,73 @@
+import unittest
+from unittest.mock import Mock, patch
+import pygame as pg
+from backend.GameManager import GameManager 
+from frontend.BoardGUI import BoardGUI 
+
+
+class TestBoardGUI(unittest.TestCase):
+    def setUp(self):
+       
+        self.mock_game = Mock()
+        self.mock_game.screen = Mock()
+        
+        pg.image.load = Mock()
+        pg.transform.smoothscale = Mock(return_value=Mock()) 
+       
+        self.board_size = 8  
+        self.win_size = 640  
+        self.total_pieces = 16
+        self.board_gui = BoardGUI(self.mock_game, self.win_size, self.board_size, self.total_pieces)
+
+    def tearDown(self):
+        # Auto cleanup done by unittest
+        pass
+
+    def test_initialization(self):
+        self.assertEqual(self.board_gui.cell_size, self.win_size // self.board_size)
+        self.assertIsInstance(self.board_gui.backend_game, GameManager)
+        self.assertEqual(self.board_gui.count, 0)
+
+    @patch('pygame.image.load')
+    @patch('pygame.transform.smoothscale')
+    def test_get_scaled_image(self, mock_smoothscale, mock_load):
+        mock_image = Mock()
+        mock_load.return_value = mock_image
+        mock_smoothscale.return_value = mock_image
+
+        # Call the method with a mock path and resolution and assert the calls were made correctly
+        path = 'resources/board/board.png'
+        resolution = [100, 100]
+        image = self.board_gui.get_scaled_image(path, resolution)
+
+        mock_load.assert_called_once_with(path)
+        mock_smoothscale.assert_called_once_with(mock_image.convert_alpha(), resolution)
+        self.assertEqual(image, mock_image)
```

Comment:
The true test for `get_scaled_image` should be that the returned image is the size of the resolution provided to the function.

Reply:
We would also want a test for a negative path. If image.get_size() is not equal to resoultion then scaled_image.get_size() doesnt equal image.get_size()

Reply:
Also Im not sure if you need to mock out the image and function here. Since we have the actual image path we can use and we can use the `pygame.transform.smoothscale` I think we can just use the actual create the test without any mocks.